### PR TITLE
IPFS sidecar

### PR DESCRIPTION
**Pull request by folex**

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,20 @@
+{
+    "[dockerfile]": {
+        "editor.quickSuggestions": {
+            "strings": true
+        },
+        "editor.formatOnSave": false
+    },
+    "shellformat.effectLanguages": [
+        "shellscript",
+        // "dockerfile",
```
Comment (justprosh) [inline .vscode/settings.json:10]: is it ok?

File: .vscode/settings.json [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+        "dotenv",
+        "hosts",
+        "jvmoptions",
+        "ignore",
+        "gitignore",
+        "properties",
+        "spring-boot-properties",
+        "azcli"
+    ]
+}
```
Comment (justprosh) [inline .vscode/settings.json:20]: newline missing

File: s6/root/defaults/ipfs_preshared_swarm.key [hunk 3]
```diff
@@ -0,0 +1,3 @@
+/key/swarm/psk/1.0.0/
+/base16/
+a3ff4566ee853b0cf9d5fa7346f91f9c7cde5c6efeaba94f90ad049c1e30267f
```
Comment (justprosh) [inline s6/root/defaults/ipfs_preshared_swarm.key:3]: ditto

File: s6/root/etc/logrotate.d/nginx [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+        weekly
+        rotate 14
+        compress
+        delaycompress
+        nodateext
+        notifempty
+        missingok
+        sharedscripts
+        postrotate
+                s6-svc -h /var/run/s6/services/nginx
```
Comment (justprosh) [inline s6/root/etc/logrotate.d/nginx:11]: maybe it should be aligned with other lines

File: .vscode/settings.json (hunk 1 above)
Comment (folex) [reply .vscode/settings.json:10]: yes

File: .vscode/settings.json (hunk 2 above)
Comment (folex) [reply .vscode/settings.json:20]: that's how VSCode formats it :(

File: s6/root/etc/logrotate.d/nginx (hunk 4 above)
Comment (folex) [reply s6/root/etc/logrotate.d/nginx:11]: nope, it's a command to run after log rotate. I'll remove nginx for now

**Pull request merged**